File: packages/typescript/examples/quickstart.ts
/**
 * Quickstart — log in to Instagram and publish a post.
 *
 * Usage:
 *   1. Copy ../../.env.example to .env at the repo root, fill in
 *      BROWSERBASE_*, ANTHROPIC_API_KEY, and STAGEHAND_SOCIAL_INSTAGRAM_*.
 *   2. From the repo root: npx tsx packages/typescript/examples/quickstart.ts
 */

import "dotenv/config";
import { SocialSDK } from "@stagehand-social/sdk";

async function main() {
  const sdk = await SocialSDK.create({
    browserbase: {
      apiKey: process.env.BROWSERBASE_API_KEY!,
      projectId: process.env.BROWSERBASE_PROJECT_ID!,
    },
    llm: {
      provider: "anthropic",
      apiKey: process.env.ANTHROPIC_API_KEY!,
    },
    headless: false, // flip to true for CI
  });

  const ig = sdk.platforms.instagram;

  console.log("Logging in…");
  await ig.login();

  console.log("Posting…");
  await ig.post({
    text: "Hello from stagehand-social-sdk!",
    media: ["./examples/sample.jpg"],
    hashtags: ["sdk", "automation"],
  });

  console.log("Done.");
  await sdk.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: packages/python/examples/quickstart.py
"""Quickstart — log in to Instagram and publish a post.

Usage:
  1. Copy ../../.env.example to .env at the repo root, fill in keys.
  2. From the repo root:
       cd packages/python && pip install -e .
       python examples/quickstart.py
"""

from __future__ import annotations

import asyncio
import os

from dotenv import load_dotenv

from stagehand_social import (
    BrowserbaseConfig,
    LLMConfig,
    PostInput,
    SocialSDK,
    SocialSDKConfig,
)


async def main() -> None:
    load_dotenv()
    sdk = await SocialSDK.create(
        SocialSDKConfig(
            browserbase=BrowserbaseConfig(
                api_key=os.environ["BROWSERBASE_API_KEY"],
                project_id=os.environ["BROWSERBASE_PROJECT_ID"],
            ),
            llm=LLMConfig(
                provider="anthropic",
                api_key=os.environ["ANTHROPIC_API_KEY"],
            ),
            headless=False,
        )
    )

    ig = sdk.platforms.instagram

    print("Logging in…")
    await ig.login()

    print("Posting…")
    await ig.post(
        PostInput(
            text="Hello from stagehand-social-sdk!",
            media=["./examples/sample.jpg"],
            hashtags=["sdk", "automation"],
        )
    )

    print("Done.")
    await sdk.close()


if __name__ == "__main__":
    asyncio.run(main())
